com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -1090,10 +1090,10 @@ private void Initialize()
         /// <param name="posIn"></param> new position to move to.  Can be null
         /// <param name="rotIn"></param> new rotation to rotate to.  Can be null
         /// <param name="scaleIn">new scale to scale to. Can be null</param>
-        /// <param name="shouldGhostsInterpolate">Should other clients interpolate this change or not. True by default</param>
+        /// <param name="isNotTeleporting">When true (the default) the <see cref="NetworkObject"/> will not be teleported. When false the <see cref="NetworkObject"/> will teleport/apply the parameters provided immediately.</param>
         /// new scale to scale to.  Can be null
         /// <exception cref="Exception"></exception>
-        public void SetState(Vector3? posIn = null, Quaternion? rotIn = null, Vector3? scaleIn = null, bool shouldGhostsInterpolate = true)
+        public void SetState(Vector3? posIn = null, Quaternion? rotIn = null, Vector3? scaleIn = null, bool isNotTeleporting = true)
```

Comment:
I recognize this parameter has to default to true for backward compatibility, but I'm not a big fan of negation in parameter names. I think it's confusing because of the double negative where `false` indicates `is not not teleporting`. I wonder if a better name might be available, like maybe

`isNormalMovement`
`isNonTeleportMovement` - still has a negative in it but makes a little more linguistic sense

In an ideal world, this would have been a nice place for an enum

```enum MovementType
{
    Normal,
    Teleport
}```

Reply:
Yeah.. it was a conundrum indeed... it ---should--- have been `isTeleporting` but as you said it breaks things.
The issue is that this parameter really translates to "Teleporting or not"... so any name that doesn't include "teleport" will lead to potential confusion.. I am not happing with the name either... 😿 

`teleportDisabled` ?